### ISSUE TITLE
wxPreferencesEditor isn't closed using CMD+W shortcut on macOS

### ISSUE THREAD

**Issue by stefanfis** [wxWidgets/wxWidgets#22863]
On macOS, the standard preferences dialog (opened by CMD+COMMA or "AppNameMenu > Preferences...") should get closed using the default close window keyboard shortcut CMD+W. At least this is how several native macOS applications work, e.g. Safari, Photos, 1Password7, TextMate, Excel, App Store, Preview all close their preferences dialog using CMD+W. (Some of them even close on pressing ESC, but that's not what should happen.)

The wxWidgets sample application "preferences" uses wxPreferencesEditor to show a native preferences dialog, but that dialog can only be closed by pressing the small red dot in the top window border.

To Reproduce:
1. Open "preferences" sample application on macOS
2. Press CMD+COMMA to open the native preferences dialog
3. Press CMD+W to close that dialog
4. Nothing happens, i.e. the dialog isn't closed

- wxWidgets version: 3.2
- wxWidgets port: wxOSX
- OS: macOS 12.6 "Monterey"

**Comment by vadz** [comment]
AFAIU you need to have a "Close" item in the menu under macOS for the accelerator to work. I don't know how to set this up, any help with fixing this would be welcome.

**Comment by stefanfis** [comment]
Indeed, for the _preferences_ sample this is just a matter of adding an accelerator and handling the _wxID_CLOSE_ menu event. I've attached a patch file that does just this. Tested only on macOS 12.6 with wxWidgets 3.2.0.

For my DocView based application, things seem to be more complicated, but that's another issue.

[preferences.cpp.patch.txt](https://github.com/wxWidgets/wxWidgets/files/9763739/preferences.cpp.patch.txt)
